<launch>
    <arg name="x_pos" default="0.0"/>
    <arg name="y_pos" default="0.5"/>
    <arg name="z_pos" default="0.0"/>
    <arg name="open_rviz" default="false"/>
    
    <!-- Include the robot's sim world -->
    <include file="$(find phase2)/launch/world_tbot.launch"/>
    <!-- Add map -->

    <!-- Turtlebot3 physical model-->
    <group ns="robot0">
        <param name="tf_prefix" value="robot0_tf" />
        <include file="$(find phase2)/launch/single_tbot.launch">
            <arg name="x_pos" value="$(arg x_pos)" />
            <arg name="y_pos" value="$(arg y_pos)" />
            <arg name="z_pos" value="$(arg z_pos)" />
            <arg name="name" value="r0" />
        </include>
        <!-- ControlNode -->
        <node name="tb3_control_node" pkg="tb3_control" type="tb3_control_node" output="screen">
            <!--param name="frame" value="$(arg frame)"/>
            <param name="worldFrame" value="$(arg worldFrame)"/>
            <rosparam command="load" file="$(find tb3_navigation)/config/utils.yaml"/-->

            <remap from="tb3_control_node/cmd_vel" to="cmd_vel"/>
            <remap from="tb3_control_node/odom" to="Odometry/filtered"/>
            <remap from="tb3_control_node/visualization_marker" to="visualization_marker"/>

        </node>
        <!-- An EKF instance that fuses odometry and IMU data and outputs an odom-frame state estimate -->
        <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true">
            <rosparam command="load" file="$(find phase2)/params/ekf_template.yaml" />
        </node> 
        <!-- BVC node -->
        <node pkg="phase2" type="phase2_node" name="phase2_node"/>
    </group>

    <!-- Turtlebot3 obstacle 1 -->
    <group ns="robot1">
        <param name="tf_prefix" value="robot1_tf" />
        <include file="$(find phase2)/launch/single_tbot.launch" >
            <arg name="x_pos" value="0.5" />
            <arg name="y_pos" value="-0.5" />
            <arg name="z_pos" value="0" />
            <arg name="name" value="obs1" />
        </include>
        <!-- simple move -->
        <node pkg="phase2" name="simple_move" type="simple_move.py" output="screen">
        <!-- Maximum robot speed in the body x direction -->
            <param name="v_max" type="double" value="0.5"/>
            <!-- Anglur range of laser scan reading considered for obstacle detection -->
            <param name="max_angle" type="int" value="300"/>
            <!-- Maximum stopping distance (along body x) -->
            <param name="range_lim" type="double" value="1"/>
            <!-- Sim update rate -->
            <param name="sample_rate" type="int" value="10"/>
        </node>
    </group>

    <!-- rviz -->
    <!-- <group if="$(arg open_rviz)"> 
        <param name="tf_prefix" value="robot0_tf" />
        <param name="robot_description" value="/robot0/robot_description" />
        <include file="$(find phase2)/launch/rviz1_tbot.launch"/>
    </group> -->
</launch>